oracle/Assets/Scripts/DelphiClient.cs

```diff
@@ -15,118 +17,126 @@ public class DelphiClient : MonoBehaviour
-
-        callback(response);
+        StartCoroutine(GetDelphiData(BASE_URL + GET_DATA_ENDPOINT));
     }
 
     /// <summary>
-    /// Gets the people count of all of the nodes given in the list of ids.
+    /// Calls a post request to the given uri with the given json.
+    /// <param name="uri">The uri to send the request to</param>
+    /// <param name="json">The json to put into the body of the request</param>
     /// </summary>
-    IEnumerator CallGetCounts(int[] nodeIds, System.Action<JSONNode> callback)
+    private IEnumerator PostJson(string uri, JSONObject json) 
     {
-        UnityWebRequest www = UnityWebRequest.Get(BASE_URL + "/get-counts" + GenerateParams("room_id", nodeIds));
-        yield return www.SendWebRequest();
-
-        if (www.isNetworkError || www.isHttpError)
+        // Cannot using UnityWebRequest.Post because it will URL encode the JSON.
+        using(UnityWebRequest www = new UnityWebRequest(uri, UnityWebRequest.kHttpVerbPOST))
         {
-            Debug.LogError(www.error);
-            yield break;
-        }
+            www.uploadHandler = new UploadHandlerRaw(Encoding.UTF8.GetBytes(json.ToString()));
+            www.SetRequestHeader("Content-Type", "application/json");
+
+            yield return www.SendWebRequest();
 
-        JSONNode response = JSON.Parse(www.downloadHandler.text);
+            if(www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogError(www.error);
+            }
 
-        callback(response);
+            yield break;
+        }
     }
 
     /// <summary>
-    /// Updates the status of the node associated with the given id.
+    /// Calls a get request to gather data from Delphi.
+    /// <param name="uri">The uri to send the request to</param>
     /// </summary>
-    IEnumerator CallToggleStatus(Node node)
+    private IEnumerator GetDelphiData(string uri)
     {
-        JSONObject requestData = new JSONObject();
-        requestData.Add("enable", (!node.Status).ToString());
+        using(UnityWebRequest www = UnityWebRequest.Get(uri))
+        {
+            www.SetRequestHeader("Content-Type", "application/json");
+            yield return www.SendWebRequest();
 
-        UnityWebRequest www = UnityWebRequest.Put(BASE_URL + "/enable/" + node.Id, requestData.ToString());
-        www.SetRequestHeader("Content-Type", "application/json");
-        yield return www.SendWebRequest();
+            if(www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogError(www.error);
+            }
+            else
+            {
+                Debug.Log("We got data!");
+                // Once we get the keywords figured out, this is where we will update the nodes on the currently selected floor.
+
+                // Update the statuses of all the graph components on this floor
+
+                // Update the room counts on this floor
+                
+                // Update the current paths available
```

Comment:
how would you grab the floor? using GameObject.Find?

Reply:
I have the ui manager in this class, which contains the current floor.